packages/rest/src/validation/request-body.validator.ts

```diff
@@ -142,13 +143,24 @@ function validateValueAgainstSchema(
     cache.set(schema, validatorMap);
   }
 
-  if (validate(body)) {
+  let validationResult = validate(body);
+  let validationErrors: AJV.ErrorObject[] = [];
+  if (isPromiseLike(validationResult)) {
```

Comment:
With [new information](https://github.com/strongloop/loopback-next/pull/4658#discussion_r380657228) at hand, the discussion changes to two points:

1. To allow async validations or not?
2. To allow custom keywords to be created or not? <= This because the PR has provision for both formats & keywords since ajv has provision to allow custom formats at options level.

Reply:
My opinions:

> To allow async validations or not?

I think allowing async validation is inevitable, let's make it happen now. We are going to publish a semver-major version of all packages in the next week or two because we have dropped support for Node.js 8.x, let's try to include the switch to async validation in that release if possible.

Either way, I think the switch to async validations deserves a standalone pull request. That way we can focus on the implications of switching to async validations and don't get distracted about formats/keywords.

> To allow custom keywords to be created or not? <= This because the PR has provision for both formats & keywords since ajv has provision to allow custom formats at options level.

I think we should allow app developers & extension developers to contribute both custom keywords and custom formats. I don't have a strong opinion on how that should be implemented, whether via standard AJV options we already support, or via additional validation config (as presented in this PR, IIUC) or via new extension-points (as I commented in https://github.com/strongloop/loopback-next/pull/4658#issuecomment-589672199).